pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2ServiceProviderMetadataResolver.java

```diff
@@ -129,17 +132,27 @@ private void writeServiceProviderMetadataToResource(final String tempMetadata) t
         final WritableResource metadataResource = configuration.getServiceProviderMetadataResource();
         if (metadataResource != null && CommonHelper.isNotBlank(tempMetadata)) {
             if (metadataResource.exists() && !configuration.isForceServiceProviderMetadataGeneration()) {
-                logger.info("Metadata file already exists at {}.", metadataResource.getFile());
+                logger.info("Metadata file already exists at {}.", metadataResource.getFilename());
```

Comment:
Log the resource, and not the file name. Not all writeable-resource implementations might in fact have a filename, and I suspect may throw an exception.
Also please adjust the log message. The resource may not always be a file as the message says.